packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -624,9 +658,10 @@ export const transformPlotlyJsonToScatterChartProps = (
         isDarkTheme,
       ) as string[] | string | undefined;
       const xValues = series.x as Datum[];
-      const isString = isStringArray(xValues);
+      const isXString = isStringArray(xValues);
       const isXDate = isDateArray(xValues);
       const isXNumber = isNumberArray(xValues);
+      const isXYearCategory = isYearArray(series.x);
```

Comment:
string case is not possible for scatter chart. It is already filtered out in declarative chart. Mark this value as false always